2014
----

[Gilbert JA, Jansson JK, and Knight R. 2014. The Earth Microbiome
project: successes and aspirations. BMC Biology*,* **12**:69 
doi:10.1186/s12915-014-0069-1](http://www.biomedcentral.com/1741-7007/12/69)
[Scott NM, Hess M, Bouskill NJ, Mason OU, Jansson JK, Gilbert JA: The
microbial nitrogen cycling potential is impacted by polyaromatic
hydrocarbon pollution of marine sediments. Front Microbiol 2014,
5.](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3971162/) [Gittel A,
Bárta J, Kohoutová I, Mikutta R, Owens S, Gilbert J, Schnecker J, Wild
B, Hannisdal B, Maerz J, Lashchinskiy N, Capek P, Santrůčková H, Gentsch
N, Shibistova O, Guggenberger G, Richter A, Torsvik VL, Schleper C,
Urich T: Distinct microbial communities associated with buried soils in
the Siberian tundra. ISME J 2014,
8:841–853.](http://www.polarresearch.at/wp-content/uploads/Gittel2014Distinct-microbial-communities-assoc.pdf)
[Mason OU, Scott NM, Gonzalez A, Robbins-Pianka A, Bælum J, Kimbrel J,
Bouskill NJ, Prestat E, Borglin S, Joyner DC, Fortney JL, Jurelevicius
D, Stringfellow WT, Alvarez-Cohen L, Hazen TC, Knight R, Gilbert JA,
Jansson JK: Metagenomics reveals sediment microbial community response
to Deepwater Horizon oil spill. ISME J
2014.](https://etd.ohiolink.edu/rws_etd/document/get/osu1366292190/inline)
[Gibbons SM, Jones E, Bearquiver A, Blackwolf F, Roundstone W, et al.
(2014) Human and Environmental Impacts on River Sediment Microbial
Communities. PLoS ONE 9(5): e97435.
doi:10.1371/journal.pone.0097435](http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0097435)
[Winston ME, Hampton-Marcell J, Zarraonaindia I, Owens SM, Moreau CS, et
al. (2014) Understanding Cultivar-Specificity and Soil Determinants of
the *Cannabis* Microbiome. PLoS ONE 9(6): e99641.
doi:10.1371/journal.pone.0099641](http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0099641)
[Piombino P, Genovese A, Esposito S, Moio L, Cutolo PP, Chambery A,
Severino V, Moneta E, Smith DP, Owens SM, Gilbert JA, Ercolini D: Saliva
from Obese Individuals Suppresses the Release of Aroma Compounds from
Wine. PLoS ONE 2014,
9:e85611.](http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0085611)

2013
----

[Fierer, N., J. Ladau, J.C. Clemente, J. Leff, S.M. Owens, K.S. Pollard,
R. Knight, J.A. Gilbert, R.L. McCulley. 2013. Reconstructing the
microbial diversity and function of pre-agricultural tallgrass prairie
soils in the United States. Science. 342:
621-624.](http://www.colorado.edu/eeb/EEBprojects/FiererLab/Fierer_etal_2013_Science_prairie.pdf)
[Gittel, A. *et al.* (2013) Distinct microbial communities associated
with buried soils in the Siberian tundra. *ISME J*,
doi:10.1038/ismej.2013.219.](http://www.nature.com/ismej/journal/vaop/ncurrent/full/ismej2013219a.html "Distinct microbial communities associated with buried soils in the Siberian tundra")
[Jansson J, Prosser J (2013) Microbiology: The life beneath our feet.
Nature 494: 40–41.
doi:10.1038/494040a.](http://www.nature.com/nature/journal/v494/n7435/full/494040a.html)
[Rubin BER, Gibbons SM, Kennedy S, Hampton-Marcell J, Owens S, et al.
(2013) Investigating the Impact of Storage Conditions on Microbial
Community Composition in Soil Samples. PLoS ONE 8(7): e70460.
doi:10.1371/journal.pone.0070460](http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0070460)

2012
----

[Gewin V: Microbes en masse: The sequencing machine. Nature 2012,
487:156–158.](http://www.nature.com/news/microbes-en-masse-the-sequencing-machine-1.10985)
[Knight R, Jansson J, Field D, Fierer N, Desai N, et al. (2012)
Unlocking the potential of metagenomics through replicated experimental
design. Nature Biotechnology 30: 513–520.
doi:10.1038/nbt.2235.](http://www.nature.com/nbt/journal/v30/n6/abs/nbt.2235.html)
[Larsen PE, Field D, Gilbert JA (2012) Predicting bacterial community
assemblages using an artificial neural network approach. Nature Methods.
doi:10.1038/nmeth.1975.](http://www.nature.com/nmeth/journal/v9/n6/abs/nmeth.1975.html)
[Caporaso JG, Lauber CL, Walters WA, Berg-Lyons D, Huntley J, et al.
(2012) Ultra-high-throughput microbial community analysis on the
Illumina HiSeq and MiSeq platforms. The ISME Journal.
doi:10.1038/ismej.2012.8.](http://www.nature.com/ismej/journal/vaop/ncurrent/full/ismej20128a.html)
[Gilbert JA, Meyer F. Modeling the Earth’s Microbiome: A real world
deliverable for microbial ecology. ASM Microbe
Magazine.](http://www.microbemagazine.org/index.php/02-2012-home/4399-modeling-the-earth-microbiome "Modeling the Earths Microbiome")

2011
----

[Karsenti E, Acinas SG, Bork P, Bowler C, De Vargas C, et al. (2011) A
Holistic Approach to Marine Eco-Systems Biology. PLoS Biol 9(10):
e1001177.
doi:10.1371/journal.pbio.1001177](http://www.plosbiology.org/article/info%3Adoi%2F10.1371%2Fjournal.pbio.1001177 "The Tara Oceans Project")
[Jansson J. 2011. Microbial ecologists are taking a metagenomics
approach to analyze complex and diverse soil microbial communities.
Microbe
Magazine](http://www.microbemagazine.org/index.php/07-2011-home/3553-towards-tera-terra-terabase-sequencing-of-terrestrial-metagenomes).
[Gilbert JA, O’Dor R, King N, Vogel T. 2011. The importance of
metagenomic surveys to microbial ecology: or why Darwin would have been
a metagenomic scientist. Microbial Informatics and
Experimentation,*1:5*](http://www.microbialinformaticsj.com/content/1/1/5 "Letter to Editor")
[Larsen PE, Collart F, Field D, Meyer F, Keegan KP, Henry CS, McGrath J,
Quinn J, Gilbert JA. 2011. **** Predicted Relative Metabolomic Turnover
(PRMT): determining metabolic turnover from a coastal marine metagenomic
dataset. Microbial Informatics and Experimentation 2011,
1:4](http://www.microbialinformaticsj.com/content/1/1/4 "Predictive Relative Metabolic Turnover")
(this is one of the modeling techniques being implemented BY the EMP for
metagenomic data analysis) [Yilmaz et al. **2011**. Minimum information
about a marker gene sequence (MIMARKS) and minimum information about any
(x) sequence (MIxS) specifications. Nature Biotechnology. 29,
415–420](http://www.nature.com/nbt/journal/v29/n5/full/nbt.1823.html "MIMARKS and MIXS standards")
(this is the standard metadata capture schema employed by the EMP)
[Gilbert JA, ODor R, Vogel T. **2011**. Survey studies are still vital
to science. Nature. 469,
162](http://www.nature.com/nature/journal/v469/n7329/full/469162a.html "Nature Correspondance").
[Gilbert JA and Dupont C. **2011**. Marine Metagenomics. Annual Review
of Marine
Science.](http://arjournals.annualreviews.org/eprint/vAxS2cEcQvIZZYnqbXE3/full/10.1146/annurev-marine-120709-142811 "Marine Metagenomics")*[Volume
3:
347-371](http://arjournals.annualreviews.org/eprint/vAxS2cEcQvIZZYnqbXE3/full/10.1146/annurev-marine-120709-142811 "Marine Metagenomics").*

2010
----

[Gilbert JA, Meyer F, Antonoploulos D, Balaji P, Brown CT, Brown CT,
Desai N, Eisen JA, Evers D, Feng W, Huson D, Jansson J, Knight R, Knight
J, Kolker E, Kostantindis K, Kostka J, Kyrpides, N, Mackelprang R,
McHardy A, Quince C, Raes J, Sczyrba A, Shade A, Stevens R. **2010**.
Meeting Report.  The Terabase Metagenomics Workshop and the Vision of an
Earth Microbiome Project. Standards in Genomic Science.
*3:3*.](http://sigen.org/index.php/sigen/article/view/sigs.1433550 "Terabase Metagenomics Workshop Report")
[Gilbert JA, Meyer F, Jansson J, Gordon J, Pace N, Tiedje J, Ley R,
Fierer N, Field D, Kyrpides N, Glockner FO, Klenk H-P, Wommack KE, Glass
E, Docherty K, Gallery R, Stevens R, Knight R. **2010**.  The Earth
Microbiome Project: Meeting report of the 1^st^ EMP meeting on sample
selection and acquisition at Argonne National Laboratory October 6^th^
2010. Standards in Genomic Science.
*3:3*.](http://sigen.org/index.php/sigen/article/view/sigs.1443528 "EMP 1st Meeting report on sample acquisition")
[Gilbert JA, Meyer F, Knight R, Field D, Kyrpides N, Yilmaz P, Wooley J.
**2010**. Meeting report: GSC M5 roundtable at the 13^th^ International
Society for Microbial Ecology meeting in Seattle, WA, USA, August
22-27^th^ 2010. Standards in Genomic Science.
*3:3*.](http://sigen.org/index.php/sigen/article/view/sigs.1333437 "ISME13 Rountable meeting report")
[<span>Gilbert JA</span><span>, Meyer F, Field D, Schriml LM, Garrity
GM. **2010**. Metagenomics: A foundling finds its feet. Standards in
Genomic Science. *Vol 3
(2)*</span>](http://journals.sfu.ca/sigen/index.php/sigen/article/viewArticle/sigs.1213842/400 "Metagenomics editorial")
[Glass E, Meyer F, Gilbert JA, Field D, Hunter S, Kottman R, Kyrpides N,
Sansone S, Schriml L, Sterk P, White O , Wooley J. **2010**. Meeting
report from the Genomics Standards Consortium (GSC) workshop 10.
Standards in Genomic Science*.
3:3*.](http://sigen.org/index.php/sigen/article/view/sigs.1423520 "GSC10 meeting report")
 
